woocommerce/woocommerce-gutenberg-products-block
# Add global style support to Mini Cart (button)

This is a part of #4965.

This PR:
- Adds text and background color support to the Mini Cart button.
- Adds option to set the button transparent (default to `true`).
- By default, the text color affects total price, cart amount, and the cart icon color. The badge color doesn't change until the background color is set (black text and border, white background, and box-shadow).
- When the background color is set, the badge background will change and the box-shadow style will be disabled. The color of badge text/border will inherit the text color as well.

Notes: The current limitation of this PR is global style doesn't work for background.

Updates: The global background support was added using `__experimentalSelector`. It's possible to not use experimental features but in my discovery, it only works for the PHP side. See [this](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/try/global-background-style-mini-cart/src/BlockTypes/MiniCart.php#L259-L271).

### Screenshots

https://user-images.githubusercontent.com/5423135/140872646-c4e10cd5-e474-4d22-80a7-c6ed23ef6425.mov



### Testing

How to test the changes in this Pull Request:

Follow the screencast above or:

0. Check out this and build the assets.
1. Add Mini Cart block to a page.
2. Set the text color, see it changes in the Editor.
3. Set the background color, see it changes in the editor.
4. Toggle transparent background, see it changes in the editor.
5. Save the page, see the style on the front-end.
6. Clear all customization for the block.
7. Set the global style to Mini Cart block using the Site Editor.
8. Edit the page created in step one, clear all colors. See the global style applies.
9. Set text color for the Mini cart block, see the text color change, but the background still inherits from the global style.
10. Toggle the transparent button option, see it works well with the global style.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
